core/src/subgraph/runner.rs

```diff
@@ -144,12 +197,17 @@ where
 
             let block_stream_canceler = CancelGuard::new();
             let block_stream_cancel_handle = block_stream_canceler.handle();
+            // TriggerFilter needs to be rebuilt eveytime the blockstream is restarted
+            self.build_filter();
 
-            let mut block_stream =
-                new_block_stream(&self.inputs, &self.ctx.filter, &self.metrics.subgraph)
-                    .await?
-                    .map_err(CancelableError::Error)
-                    .cancelable(&block_stream_canceler, || Err(CancelableError::Cancel));
+            let mut block_stream = new_block_stream(
+                &self.inputs,
+                self.ctx.filter.as_ref().unwrap(), // Safe to unwrap as we just called `build_filter` in the previous line
```

Comment:
we would also still have the filter here so maybe we could just pass that in instead of unwrapping, mostly a read improvement

Reply:
To prevent the unwrapping, i would need to clone the filter to once to assign to `self.ctx.filter` and then to pass it to be blockstream, so just going with unwrap for now